v1

skndash96 opened the pull request.

AritraDey-Dev commented on package-lock.json:
```diff
@@ -1,4858 +0,0 @@
-{
```
No need to remove `package-lock.json`.This will create a issue while automatic deployment if we do integrate `jenkins` to this.

skndash96 replied:
added npm lock file in the latest commit

AritraDey-Dev commented on env.ex:
Do not remove `env.ex ` .

skndash96 replied:
added env.example in the latest commit

AritraDey-Dev commented on pnpm-lock.yaml:
Same for this file.Will create a issue in my local env as i am using `pnpm` most of the cases.We can keep this.

AritraDey-Dev commented on controllers/paymentController.js:
```diff
@@ -1,31 +1,71 @@
+    }
+  
+    const razorPayOrderId = order.paymentInfo.id
+  
```
I think razorpay hook is working properly right.Just let me know if you have tested this locally.

skndash96 replied:
there's no webhook setup. right now how it works is-
user pays in the browser via razorpay. after payment is done/fails, razorpay calls the callback function.
which makes a post request to the backend at `/api/payment/verify`, it checks the razorpay payment status and updates the mongodb database.
but this is not reliable. so we've to setup a webhook to upate the mongodb database.

AritraDey-Dev replied:
Could you quickly complete this.As I think only you need to update db schema and update it whenever there is some payment.

AritraDey-Dev reviewed (commented):
It will be good if you can give some screenshots of postman or calling the api in any way you prefer.It will be enough if you just post the screenshot of the `routes`  you modified not all the routes.

skndash96 commented:
## ✅ Code Changes Summary

- **Added** `constants/categories.js`  
  - Contains all fixed product category values.

- **Product Model Updates**  
  - Added `createdAt` field.

- **Order Model Updates**  
  - Added `createdAt` field (required).  
  - Updated `orderItems`:  
    - Removed: `image`  
    - Added: `category`, `description`, `images` (array of strings)

- **Product Review Model Update**  
  - Added `createdAt` field (required)

- **Controller Changes**  
  - `getAllProducts`: added support for `category` as a search query param.  
  - `verifyOrder`: checks Razorpay payment status and updates order in MongoDB.

---

## 🔁 Modified Routes

### `/api/products`
- **Method:** `GET`
- **Query Params:**
  - `category` *(optional)* — Filter products by category.
- **Description:**
  - Returns all products or filters by category if provided.

### `/api/orders`
- **Method:** `POST`
- **Body Changes:**
  - **Removed:** `name`, `email` (from root level)
  - **Added:** `user` object → `{ name, email }`
  - **Note:** Also reflects changes in `orderItems` structure (see model updates above).

---

## 🆕 New Routes

### `/api/products/categories`
- **Method:** `GET`
- **Description:**
  - Returns all available product categories.

### `/api/payment/verify`
- **Method:** `POST`
- **Body:**
  ```json
  {
    "orderId": "string (required)"
  }
- **Description:** Verifies Razorpay payment and updates the order in MongoDB

AritraDey-Dev commented on lib/stripe.js:
```diff
@@ -0,0 +1,15 @@
+const Stripe = require('stripe');
```
we are not using `stripe` right.Then why this file is here.can we remove this?
